sdk/instrumentation/opentelemetry-instrumentation-azure-sdk/src/instrumenter.ts

```diff
@@ -21,11 +29,23 @@ export class OpenTelemetryInstrumenter implements Instrumenter {
     name: string,
     spanOptions: InstrumenterSpanOptions
   ): { span: TracingSpan; tracingContext: TracingContext } {
-    const span = trace
-      .getTracer(spanOptions.packageName, spanOptions.packageVersion)
-      .startSpan(name, toSpanOptions(spanOptions));
-
     let ctx = spanOptions?.tracingContext || context.active();
+    let span: Span;
+
+    if (envVarToBoolean("AZURE_TRACING_DISABLED")) {
+      // disable only our spans but not any downstream spans
+      span = trace.wrapSpanContext(INVALID_SPAN_CONTEXT);
+    } else {
+      // Create our span
+      span = trace
+        .getTracer(spanOptions.packageName, spanOptions.packageVersion)
+        .startSpan(name, toSpanOptions(spanOptions), ctx);
+
+      if (envVarToBoolean("AZURE_HTTP_TRACING_DISABLED") && name.toUpperCase().startsWith("HTTP")) {
+        // disable downstream spans
+        ctx = suppressTracing(ctx);
```

Comment:
besides setting the suppressTracing context, don't we also have to return `trace.wrapSpanContext(INVALID_SPAN_CONTEXT);` as the span instead of creating it above?

Reply:
The idea for this branch (the else branch) is that we do want to create and record _our_ spans but we do not want to record child spans of our HTTP span (which are likely generated by some other auto-instrumentation like `HttpInstrumentation` and are duplicated) so in this particular case we want to:
- Create our spans
- If AZURE_HTTP_TRACING_DISABLED is true we suppress creation of child spans past our http policy. 
Although I realize the env var name is slighly confusing, it is the one we settled on but maybe we should reconsider the name

Reply:
The scenario now makes perfect sense but yes I was confused by the name.

Maybe `AZURE_HTTP_TRACING_CHILDREN_DISABLED` or something?

Reply:
We discussed offline but just for posterity I created https://github.com/Azure/azure-sdk-for-js/issues/20834 to discuss the name of the environment variable next week.